### PR TITLE
Added sync-labels workflow for GitHub Actions

### PR DESCRIPTION
**Pull request by patcon**
Possible application for https://github.com/hyphacoop/organizing/issues/145#issuecomment-554188198

Open to the view that maybe this is too much magic :)

cc: @hyphacoop/infra-ops-wg

### CODE REVIEW

File: .github/workflows/sync-labels.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      - master
+
+jobs:
+  label-sync:
+    name: Install & Run
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_API_TOKEN: ${{ secrets.GITHUB_API_TOKEN }}
+      LABEL_REPO_SOURCE: hyphacoop/organizing
+      LABEL_REPO_TARGETS: hyphacoop/organizing-private hyphacoop/handbook hyphacoop/shortlinks hyphacoop/hypha.coop hyphacoop/chatbot hyphacoop/worker-coop-scripts hyphacoop/spreadsheet2shortlinks hyphacoop/vision.hypha.coop
```
Comment (patcon) [inline .github/workflows/sync-labels.yml]: Whitelist of all repos we'd want to sync labels in `hyphacoop/organizing` into.

File: .github/workflows/sync-labels.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  push:
+    branches:
+      - master
+
+jobs:
+  label-sync:
+    name: Install & Run
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_API_TOKEN: ${{ secrets.GITHUB_API_TOKEN }}
```
Comment (patcon) [inline .github/workflows/sync-labels.yml]: This is using my personal access token. We'd probably want to give it a less permissive token from @hyphacoop-bot

File: .github/workflows/sync-labels.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      LABEL_REPO_SOURCE: hyphacoop/organizing
+      LABEL_REPO_TARGETS: hyphacoop/organizing-private hyphacoop/handbook hyphacoop/shortlinks hyphacoop/hypha.coop hyphacoop/chatbot hyphacoop/worker-coop-scripts hyphacoop/spreadsheet2shortlinks hyphacoop/vision.hypha.coop
+    steps:
+      - name: Setup PHP
+        uses: shivammathur/setup-php@master
+        with:
+          php-version: '7.3'
+      - run: |
+          composer require vanilla/github-sync
```
Comment (patcon) [inline .github/workflows/sync-labels.yml]:
```suggestion
          # See: https://github.com/vanilla/github-sync
          composer require vanilla/github-sync
```

File: .github/workflows/sync-labels.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  label-sync:
+    name: Install & Run
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_API_TOKEN: ${{ secrets.GITHUB_API_TOKEN }}
+      LABEL_REPO_SOURCE: hyphacoop/organizing
+      LABEL_REPO_TARGETS: hyphacoop/organizing-private hyphacoop/handbook hyphacoop/shortlinks hyphacoop/hypha.coop hyphacoop/chatbot hyphacoop/worker-coop-scripts hyphacoop/spreadsheet2shortlinks hyphacoop/vision.hypha.coop
+    steps:
+      - name: Setup PHP
+        uses: shivammathur/setup-php@master
```
Comment (patcon) [inline .github/workflows/sync-labels.yml]:
```suggestion
        # See: https://github.com/shivammathur/setup-php
        uses: shivammathur/setup-php@master
```

File: .github/workflows/sync-labels.yml [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+      LABEL_REPO_SOURCE: hyphacoop/organizing
+      LABEL_REPO_TARGETS: hyphacoop/organizing-private hyphacoop/handbook hyphacoop/shortlinks hyphacoop/hypha.coop hyphacoop/chatbot hyphacoop/worker-coop-scripts hyphacoop/spreadsheet2shortlinks hyphacoop/vision.hypha.coop
+    steps:
+      - name: Setup PHP
+        # See: https://github.com/shivammathur/setup-php
+        uses: shivammathur/setup-php@master
+        with:
+          php-version: '7.3'
+      - run: |
+          # See: https://github.com/vanilla/github-sync
```
Comment (patcon) [inline .github/workflows/sync-labels.yml]:
```suggestion
```

File: .github/workflows/sync-labels.yml [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+      GITHUB_API_TOKEN: ${{ secrets.GITHUB_API_TOKEN }}
+      LABEL_REPO_SOURCE: hyphacoop/organizing
+      LABEL_REPO_TARGETS: hyphacoop/organizing-private hyphacoop/handbook hyphacoop/shortlinks hyphacoop/hypha.coop hyphacoop/chatbot hyphacoop/worker-coop-scripts hyphacoop/spreadsheet2shortlinks hyphacoop/vision.hypha.coop
+    steps:
+      - name: Setup PHP
+        # See: https://github.com/shivammathur/setup-php
+        uses: shivammathur/setup-php@master
+        with:
+          php-version: '7.3'
+      - run: |
```
Comment (patcon) [inline .github/workflows/sync-labels.yml]:
```suggestion
      # See: https://github.com/vanilla/github-sync
      - run: |
```

File: .github/workflows/sync-labels.yml [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+      - master
+
+jobs:
+  label-sync:
+    name: Install & Run
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_API_TOKEN: ${{ secrets.GITHUB_API_TOKEN }}
+      LABEL_REPO_SOURCE: hyphacoop/organizing
+      LABEL_REPO_TARGETS: hyphacoop/organizing-private hyphacoop/handbook hyphacoop/shortlinks hyphacoop/hypha.coop hyphacoop/chatbot hyphacoop/worker-coop-scripts hyphacoop/spreadsheet2shortlinks hyphacoop/vision.hypha.coop
```
Comment (patcon) [inline .github/workflows/sync-labels.yml]:
```suggestion
      # Note: `hyphacoop-bot` user must also be given `write` access to repos via @hyphacoop/robots team.
      # See: https://github.com/orgs/hyphacoop/teams/robots/repositories
      LABEL_REPO_TARGETS: hyphacoop/organizing-private hyphacoop/handbook hyphacoop/shortlinks hyphacoop/hypha.coop hyphacoop/chatbot hyphacoop/worker-coop-scripts hyphacoop/spreadsheet2shortlinks hyphacoop/vision.hypha.coop
```

File: .github/workflows/sync-labels.yml (hunk 2 above)
Comment (patcon) [reply .github/workflows/sync-labels.yml]:
```suggestion
      # Personal access token for `hyphacoop-bot` with `repo` permission
      GITHUB_API_TOKEN: ${{ secrets.GITHUB_API_TOKEN }}
```

**Comment by patcon** [comment]
Looking for feedback on this, as it should get our labels sync'ing automatically and non-destructively  from `hyphacoop/organizing`. Fuller context in link in top comment.

**Comment by dcwalk** [comment]
I'm fine with this, but I do think this seems like a touch of magic that will be prone to being forgotten

**Comment by patcon** [comment]
@dcwalk good point. what if we added this, so it doesn't become forgotten arcane magic, but instead posts to an issue to make it's actions known?

```
# Posts comment to https://github.com/hyphacoop/organizing/issues/145
NOTIFIED_REPO=hyphacoop/organizing
NOTIFIED_ISSUE=145
EXPLAINING_LINK=https://example.com
curl --request POST \
  --data '{"body": "Labels have been synced!\nSee: $EXPLAINING_LINK"}' \
  --header "Authorization: token $GITHUB_API_TOKEN"
  "https://api.github.com/repos/${NOTIFIED_REPO}/issues/${NOTIFIER_ISSUE}/comments"
```

**Comment by patcon** [comment]
How's this look @dcwalk?

https://github.com/hyphacoop/demo-repo/issues/2

**Comment by patcon** [comment]
> does the Notify issue have to stay open? Or will it get re-opened if a comment is posted to it?

Nope, issue can be closed or set to another or whatever. Thanks! If it's annoying, totes happy to just disable that bit :)

**Comment by patcon** [comment]
ok! working after a small bugfix on master. All the `hyphacoop/organizing` labels are available in the other repos!

**Pull request merged**